### DISCUSSION TITLE
API: How to find agents of one given site or client?

### DISCUSSION THREAD

**Discussion by FuchsEDV-Michael-Fuchs** [amidaware/tacticalrmm#942] [Q&A]
Hi all, 
trying to use the api - I'm confused about how i can get agents for a certain client / site:

I'm able to list all clients (/clients/) - or one client (/client/{id}/) and the clients sites (/clients/{id}/sites/)
Also I can get all agents (/agents/) -> but I really get ALL agents - and there is no client-id or site-id in the information -> only names...

What I'm looking for is something like 
/clients/{id}/agents/   for getting agent-ids or
/clients/{id}/{siteid}/agents/ 

Maybe someone can explain to me, how i can get agents of one client or one site...

Thank you!
Best Michael

**Comment by wh1te909** [comment]
hi use these endpoints (replace with client and site ids)
```
/agents/?client=2
/agents/?site=6
```
